BitwardenShared/UI/Auth/IntroCarousel/IntroCarouselProcessor.swift

```diff
@@ -22,22 +29,45 @@ class IntroCarouselProcessor: StateProcessor<IntroCarouselState, IntroCarouselAc
     ///
     init(
         coordinator: AnyCoordinator<AuthRoute, AuthEvent>,
+        services: Services,
         state: IntroCarouselState
     ) {
         self.coordinator = coordinator
+        self.services = services
         super.init(state: state)
     }
 
     // MARK: Methods
 
+    override func perform(_ effect: IntroCarouselEffect) async {
+        switch effect {
+        case .createAccount:
+            let isEmailVerificationEnabled: Bool = await services.configService.getFeatureFlag(
+                .emailVerification,
+                isPreAuth: true
+            )
+            if isEmailVerificationEnabled {
+                coordinator.navigate(to: .startRegistration, context: self)
+            } else {
+                coordinator.navigate(to: .createAccount)
+            }
+        }
+    }
+
     override func receive(_ action: IntroCarouselAction) {
         switch action {
-        case .createAccount:
-            coordinator.navigate(to: .createAccount)
         case let .currentPageIndexChanged(newValue):
             state.currentPageIndex = newValue
         case .logIn:
             coordinator.navigate(to: .landing)
         }
     }
 }
+
+// MARK: - StartRegistrationDelegate
+
+extension IntroCarouselProcessor: StartRegistrationDelegate {
+    func didChangeRegion() async {
```

Comment:
is this needed then if theres no-op?

Reply:
I was getting an error if I didn't specify a delegate:

https://github.com/bitwarden/ios/blob/52283cf9be95429ea33d6f4f02df89397a6be78a/BitwardenShared/UI/Auth/AuthCoordinator.swift#L736-L738

I guess this is better than failing silently if you needed the delegate and didn't set it?

Reply:
yeah, this makes sense to me.  thanks for explaining!